plugins/inputs/diskio/diskio_linux.go

```diff
@@ -37,9 +35,22 @@ func (s *DiskIO) diskInfo(devName string) (map[string]string, error) {
 		return ic.values, nil
 	}
 
-	major := unix.Major(uint64(stat.Rdev))
-	minor := unix.Minor(uint64(stat.Rdev))
-	udevDataPath := fmt.Sprintf("%s/b%d:%d", udevPath, major, minor)
+	var udevDataPath string
+	if ok && len(ic.udevDataPath) > 0 {
+		// We can reuse the udev data path from a "previous" entry.
+		// This allows us to also "poison" it during test scenarios
+		udevDataPath = ic.udevDataPath
+	} else {
+		major := unix.Major(uint64(stat.Rdev))
+		minor := unix.Minor(uint64(stat.Rdev))
+		udevDataPath = fmt.Sprintf("/run/udev/data/b%d:%d", major, minor)
+
+		f, err := os.Open(udevDataPath)
+		defer f.Close()
+		if err != nil {
+			udevDataPath = fmt.Sprintf("/dev/.udev/db/block:%s", devName)
```

Comment:
Should we also check if we can open that path and error out otherwise?

Reply:
Oops, looks like I resolved this without actually posting the response:

The original path presence check is still there, now on line 63 (not shown in the default diff on github)

Reply:
Could you please move this test up to this location!? It's hard for a reviewer to jump through the code like this... :-)

Reply:
I considered keeping the code changes minimal more important as that would reduce the amount of risk in changing code paths, basically the whole idea of this patch was to assume the new udev would be present but gracefully handle the fallback to old /dev/.udev/db/block format which was unhandled before.